addons/stock/models/stock_picking.py

```diff
@@ -942,6 +942,24 @@ def button_validate(self):
             pickings_to_backorder = self
         pickings_not_to_backorder.with_context(cancel_backorder=True)._action_done()
         pickings_to_backorder.with_context(cancel_backorder=False)._action_done()
+
+        if self.user_has_groups('stock.group_reception_report') \
+                and self.user_has_groups('stock.group_auto_reception_report') \
+                and self.filtered(lambda p: p.picking_type_id.code != 'outgoing'):
+            lines = self.move_lines.filtered(lambda m: m.product_id.type == 'product' and m.state != 'cancel' and m.quantity_done and not m.move_dest_ids)
+            if lines:
+                # don't show reception report if all already assigned/nothing to assign
+                wh_location_ids = self.env['stock.location'].search([('id', 'child_of', self.picking_type_id.warehouse_id.view_location_id.id), ('location_id.usage', '!=', 'supplier')]).ids
+                if self.env['stock.move'].search_count([
```

Comment:
Checking the existence is faster using search + limit = 1 (psql stops directly when it finds one match) vs search count (which will check condition on every record and return a count).